packages/vx-zoom/src/Zoom.tsx

```diff
@@ -83,119 +159,119 @@ class Zoom extends React.Component {
-    });
-  }
+  setTransformMatrix = (transformMatrix: TransformMatrix) => {
+    this.setState(prevState => ({
+      transformMatrix: this.constrain(transformMatrix, prevState.transformMatrix),
+    }));
+  };
 
-  invert() {
+  invert = () => {
     return inverseMatrix(this.state.transformMatrix);
-  }
+  };
 
-  toStringInvert() {
+  toStringInvert = () => {
     const { translateX, translateY, scaleX, scaleY, skewX, skewY } = this.invert();
     return `matrix(${scaleX}, ${skewY}, ${skewX}, ${scaleY}, ${translateX}, ${translateY})`;
-  }
+  };
 
-  constrain(transformMatrix, prevTransformMatrix) {
-    const { scaleXMin, scaleXMax, scaleYMin, scaleYMax } = this.props;
-    const { scaleX, scaleY } = transformMatrix;
-    const shouldConstrainScaleX = scaleX > scaleXMax || scaleX < scaleXMin;
-    const shouldConstrainScaleY = scaleY > scaleYMax || scaleY < scaleYMin;
+  constrain =
+    this.props.constrain ||
+    ((transformMatrix: TransformMatrix, prevTransformMatrix: TransformMatrix) => {
+      const { scaleXMin, scaleXMax, scaleYMin, scaleYMax } = this.props;
+      const { scaleX, scaleY } = transformMatrix;
+      const shouldConstrainScaleX = scaleX > scaleXMax! || scaleX < scaleXMin!;
+      const shouldConstrainScaleY = scaleY > scaleYMax! || scaleY < scaleYMin!;
 
-    if (shouldConstrainScaleX || shouldConstrainScaleY) {
-      return prevTransformMatrix;
-    }
-    return transformMatrix;
-  }
+      if (shouldConstrainScaleX || shouldConstrainScaleY) {
+        return prevTransformMatrix;
+      }
+      return transformMatrix;
+    });
 
-  dragStart(event) {
+  dragStart = (event: React.MouseEvent) => {
     const { transformMatrix } = this.state;
     const { translateX, translateY } = transformMatrix;
     this.startPoint = localPoint(event);
     this.startTranslate = { translateX, translateY };
     this.setState({ isDragging: true });
-  }
+  };
 
-  dragMove(event) {
-    if (!this.state.isDragging) return;
+  dragMove = (event: React.MouseEvent) => {
+    if (!this.state.isDragging || !this.startPoint || !this.startTranslate) return;
     const currentPoint = localPoint(event);
     const dx = -(this.startPoint.x - currentPoint.x);
     const dy = -(this.startPoint.y - currentPoint.y);
     this.setTranslate({
       translateX: this.startTranslate.translateX + dx,
       translateY: this.startTranslate.translateY + dy,
     });
-  }
+  };
 
-  dragEnd(/** event */) {
+  dragEnd = () => {
     this.startPoint = undefined;
     this.startTranslate = undefined;
     this.setState({ isDragging: false });
-  }
+  };
 
-  handleWheel(event) {
-    const { passive } = this.props;
+  handleWheel = (event: React.WheelEvent | WheelEvent) => {
+    const { passive, wheelDelta } = this.props;
     if (!passive) event.preventDefault();
-    const { wheelDelta } = this.props;
     const point = localPoint(event);
-    const { scaleX, scaleY } = wheelDelta(event);
+    const { scaleX, scaleY } = wheelDelta!(event);
```

Comment:
it can be just `wheelDelta`, cause you have a default props

Reply:
this actually needs the `!` because we don't have any defaultProps helpers so it doesn't know they are defined.